src/utils/errors.ts

```diff
@@ -12,8 +12,10 @@ export class ArgumentTypeError extends Error {
 
 export class TypedError extends Error {
     type: string;
-    constructor(message?: string, type?: string) {
+    transactionHash?: string;
+    constructor(message?: string, type?: string, transactionHash?: string) {
```

Comment:
I feel it makes sense to wrap `transactionHash` into object so that we can later add more context (like `accountId`) later easily.

So that error is basically created like:

`new TypedError(message, type, { transactionHash })`

resulting object field can be something like `data` or `context`

Reply:
done